db/migrate/20250330143647_add_counter_caches_to_events.rb

```diff
@@ -0,0 +1,15 @@
+class AddCounterCachesToEvents < ActiveRecord::Migration[7.1]
+  def change
+    add_column :events, :sessions_count, :integer, default: 0
+    add_column :events, :rooms_count, :integer, default: 0
+    add_column :events, :timeslots_count, :integer, default: 0
+
+    reversible do |dir|
+      dir.up do
+        say_with_time "Updating Event counter caches..." do
+          Event.find_each { |event| Event.reset_counters(event.id, :sessions, :rooms, :timeslots) }
```

Comment:
I opted to just include these within each migration, to avoid having to manually run anything in the rails console post-deploy.